x-pack/plugins/ml/server/models/results_service/anomaly_charts.ts

```diff
@@ -1055,9 +1055,11 @@ export function anomalyChartsDataProvider(mlClient: MlClient, client: IScopedClu
             // differently because of how the source data is structured.
             // For rare chart values we are only interested wether a value is either `0` or not,
             // `0` acts like a flag in the chart whether to display the dot/marker.
-            // All other charts (single metric, population) are metric based and with
+            // For single metric chart, we need to pass null values to display data gaps.
+            // All other charts are metric based and with
             // those a value of `null` acts as the flag to hide a data point.
             if (
+              chartType === CHART_TYPE.SINGLE_METRIC ||
```

Comment:
As discussed offline, sounds like this is causing scheduled event markers to be displayed at y-axis `0`, where they should be plotted at y axis min:

<img width="1911" alt="Screenshot 2024-09-27 at 10 45 46" src="https://github.com/user-attachments/assets/af84ff66-a6d5-4795-9584-81619d5ebd66">

Reply:
Fixed in [c5fd024](https://github.com/elastic/kibana/pull/194119/commits/c5fd024457efa847085fb9f0cc3ae34a341651ba#diff-3a4e9994c4982e6e4d523e1b418c9d7521338b4569f1537e5fbbbcb941083b7eR483)
I've removed the filtering of `null` values for dots when there is a scheduled event at the data point, since there is no data behind the marker, it should be displayed at the bottom of the chart (`lineChartYScale.domain()[0]`)